src/components/items/ItemScreen.js

```diff
@@ -48,136 +42,134 @@ const ItemScreen = () => {
-  const itemsWithCreators = insertCreatorWithItems(itemData);
-  const item = itemsWithCreators.find(({ id }) => itemId === id);
+  const { data: item, isLoading } = useItem(itemId);
+
+  const { data: children, isLoading: childrenLoading } = useChildren(itemId);
+
+  const { data: members, isLoading: membersLoading } = useItemMembers(itemId);
 
-  const parentsIds = getParentsIdsFromPath(item?.path);
-  const parents = getItemFromIds(parentsIds);
-  const children = getChildren(itemsWithCreators, itemId);
+  const { data: parents, isLoading: parentsLoading } = useParents(itemId);
 
-  const members = parents
-    .map((parent) => {
-      const memberships = getMembershipsByItemPath(parent.path);
-      return getMembersByMemberships(memberships);
-    })
-    .flat();
+  if (isLoading || childrenLoading || membersLoading || parentsLoading) {
+    return <Loader />;
+  }
 
   const handleChange = (event, newValue) => {
     setValue(newValue);
   };
 
   return (
     <div>
-      <Box
-        justifyContent="left"
-        display="flex"
-        p={1}
-        bgcolor="background.paper"
-      >
-        <Box display="flex" flexDirection="row">
-          <Box p={2}>
-            <ItemIcon
-              size="100"
-              type={item?.type}
-              extra={item?.extra}
-              name={item?.name}
-            />
-          </Box>
-          <Box display="flex" p={2} flexDirection="column">
-            <Typography align="left" id={buildNavigationLink(item?.id)}>
-              {`Id: ${item?.id}`}
-            </Typography>
-            <Typography align="left" id={buildNavigationLink(item?.id)}>
-              {`Type: ${item?.type}`}
-            </Typography>
-            <Typography align="left" id={buildNavigationLink(item?.id)}>
-              {`Name: ${item?.name}`}
-            </Typography>
-            <Typography align="left" id={buildNavigationLink(item?.id)}>
-              {`Description: ${item?.description}`}
-            </Typography>
-          </Box>
-          <Box display="flex" p={2} flexDirection="column">
-            <Typography align="left" id={buildNavigationLink(item?.id)}>
-              {`Owner `}
-              <Link color="inherit" to={buildMemberPath(item?.creator)}>
-                {item?.owner}
-              </Link>
-            </Typography>
-            <Typography align="left" id={buildNavigationLink(item?.id)}>
-              {`Created At: ${formatDate(item?.createdAt)}`}
-            </Typography>
-            <Typography align="left" id={buildNavigationLink(item?.id)}>
-              {`Last time updated: ${formatDate(item?.updatedAt)}`}
-            </Typography>
+      <>
+        <Box
+          justifyContent="left"
+          display="flex"
+          p={1}
+          bgcolor="background.paper"
+        >
+          <Box display="flex" flexDirection="row">
+            <Box p={2}>
+              <ItemIcon
+                size="100"
+                type={item.get('type')}
```

Comment:
Perhaps it would be easier to define `type = item.get` to avoid getting it each time (same for name, etc..)